### PR TITLE
Allow any_errors_fatal to be dynamic

### LINKED ISSUE #61025: any_errors_fatal not accepting variables
##### SUMMARY

- any_errors_true always runs as "true" when trying to use a variable (any_errors_true: "{{ foo }}")
- String under setting fails as well (any_errors_fatal: "False")
- Works as intended if defined statically (any_errors_fatal: False)
##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
any_errors_fatal
##### ANSIBLE VERSION
```paste below
ansible 2.8.4
  config file = /private/etc/ansible/ansible.cfg
  configured module search path = ['/Users/jspringe/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/Cellar/ansible/2.8.4/libexec/lib/python3.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.7.4 (default, Jul  9 2019, 18:13:23) [Clang 10.0.1 (clang-1001.0.46.4)]
```

##### CONFIGURATION

```paste below
ANSIBLE_SSH_CONTROL_PATH(/private/etc/ansible/ansible.cfg) = ~/.ssh/mux-%r@%h:%p
DEFAULT_CALLBACK_PLUGIN_PATH(/private/etc/ansible/ansible.cfg) = ['/etc/ansible/plugins']
DEFAULT_CALLBACK_WHITELIST(/private/etc/ansible/ansible.cfg) = ['timer', 'profile_roles', 'task_roles']
DEFAULT_GATHERING(/private/etc/ansible/ansible.cfg) = explicit
DEFAULT_HOST_LIST(/private/etc/ansible/ansible.cfg) = ['/etc/ansible/host']
HOST_KEY_CHECKING(/private/etc/ansible/ansible.cfg) = False
```

##### OS / ENVIRONMENT

- MacOS
- RHEL 7.X

##### STEPS TO REPRODUCE
```yaml
- hosts: all
  gather_facts: false
  any_errors_fatal: "{{ fatal }}"
  vars:
    fatal: false
  tasks:
  - debug:
      msg: Fatal is set to "{{ fatal }}"

  - fail:
      msg: triggering a failure
    when: "inventory_hostname == 'localhost'"

  - debug:
      msg: I should still run on other hosts
```

##### EXPECTED RESULTS

When a variable is included and set to false, playbook should not trigger fatal_error failure. The playbook should continue as expected. 

```yaml
PLAY [all] **************************************************************************************************************************************************************************

TASK [debug] ************************************************************************************************************************************************************************
Wednesday 21 August 2019  12:23:02 -0400 (0:00:00.113)       0:00:00.113 ******
ok: [localhost] => {
    "msg": "Fatal is set to \"False\""
}
ok: [rhel-bastion-1] => {
    "msg": "Fatal is set to \"False\""
}

TASK [fail] *************************************************************************************************************************************************************************
Wednesday 21 August 2019  12:23:02 -0400 (0:00:00.050)       0:00:00.163 ******
fatal: [localhost]: FAILED! => {"changed": false, "msg": "triggering a failure"}
skipping: [rhel-bastion-1]

TASK [debug] ************************************************************************************************************************************************************************
Wednesday 21 August 2019  12:23:02 -0400 (0:00:00.041)       0:00:00.205 ******
ok: [rhel-bastion-1] => {
    "msg": "I should still run on other hosts"
}

PLAY RECAP **************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
rhel-bastion-1             : ok=2    changed=0    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0
```

##### ACTUAL RESULTS

Playbook fails on remaining hosts when a failure is triggered.

```paste below
TASK [debug] ************************************************************************************************************************************************************************
Wednesday 21 August 2019  12:11:23 -0400 (0:00:00.125)       0:00:00.125 ******
ok: [localhost] => {
    "msg": "Fatal is set to \"False\""
}
ok: [rhel-bastion-1] => {
    "msg": "Fatal is set to \"False\""
}

TASK [fail] *************************************************************************************************************************************************************************
Wednesday 21 August 2019  12:11:23 -0400 (0:00:00.043)       0:00:00.168 ******
fatal: [localhost]: FAILED! => {"changed": false, "msg": "triggering a failure"}
skipping: [rhel-bastion-1]

NO MORE HOSTS LEFT ******************************************************************************************************************************************************************

PLAY RECAP **************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
rhel-bastion-1             : ok=1    changed=0    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0
```

### PR DESCRIPTION
**Pull request by mwhahaha**
This change updates the linear and free strategies to ensure that the
value of any_errors_fatal is properly evaulated if it contains a dynamic
value. Previously if it was not set to a boolean false, it would be
true.

Fixes #61025

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

lib/ansible/plugins/strategy/free.py
lib/ansible/plugins/strategy/linear.py

### CODE REVIEW

File: lib/ansible/plugins/strategy/linear.py [hunk 1]
```diff
@@ -261,6 +261,7 @@ def run(self, iterator, play_context):
                     # will only send this task to the first host in the list.
 
                     task.action = templar.template(task.action)
+                    task.any_errors_fatal = templar.template(task.any_errors_fatal)
```
Comment (bcoca) [inline lib/ansible/plugins/strategy/linear.py]: use post_validate, since templating leaves out other processing and checks.

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/strategy/linear.py (hunk 1 above)
Comment (mwhahaha) [reply lib/ansible/plugins/strategy/linear.py]: What do you mean? I did check and it is being properly updated in post_validate but it doesn't appear to be properly evaluated at this point in the strategy execution.

File: test/integration/targets/any_errors_fatal/dynamic_playbook.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+- hosts: all
+  gather_facts: false
+  any_errors_fatal: "{{ not skip_errors }}"
```
Comment (bcoca) [inline test/integration/targets/any_errors_fatal/dynamic_playbook.yml:3]: also needs a test in which its set to a templated value that is not a valid boolean

File: lib/ansible/plugins/strategy/linear.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/plugins/strategy/linear.py]: sorry, for single value use `get_validated_value` which is the 'specific' version of post_validate

Comment (mwhahaha) [reply lib/ansible/plugins/strategy/linear.py]: got it. thanks

File: test/integration/targets/any_errors_fatal/dynamic_playbook.yml (hunk 2 above)
Comment (mwhahaha) [reply test/integration/targets/any_errors_fatal/dynamic_playbook.yml:3]: added

File: lib/ansible/plugins/strategy/linear.py (hunk 1 above)
Comment (mwhahaha) [reply lib/ansible/plugins/strategy/linear.py]: I think i figured out what you were asking for, please let me know if i'm wrong

**Review by bcoca — APPROVED** [review]

**Comment by nitzmahone** [comment]
With the change to the underlying stuff in #73908, the solution to this should be a lot easier now under the more dynamic descriptors- closing this impl since it's no longer applicable. Thanks!

**Pull request closed without merging**